### ISSUE TITLE
Allow for auth config to download vod listed for subscribers only

### ISSUE THREAD

**Issue by kereliuk** [ihabunek/twitch-dl#48]
Is there a way to do this currently?

**Comment by ihabunek** [comment]
I don't have a way to test this currently since none of the channels I subscribe to have any subscriber-only vods. If someone knows how this would be implemented, I can help out.

**Comment by fchaix** [comment]
This one has sub-onl vods : https://www.twitch.tv/usul2000

**Comment by pikess** [comment]
+1

**Comment by Wayneoween** [comment]
I'd like that, too. Critical Role has Sub-Only Videos: https://www.twitch.tv/criticalrole/videos

**Comment by ihabunek** [comment]
Here's how youtube-dl does it: https://github.com/ytdl-org/youtube-dl/blob/530f4582d011cd94986cf4d233f9fb9263f72150/youtube_dl/extractor/twitch.py#L60

Looks a bit janky but doable.

**Comment by ihabunek** [comment]
It seems that the login procedure in youtube-dl doesn't works, it hits a captcha.

```
λ youtube-dl --username *** --password *** https://www.twitch.tv/videos/1314751308
[twitch:vod] Downloading login page
[twitch:vod] Logging in
ERROR: Unable to login. Twitch said: Please enter a Login Verification Code.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 533, in extract
    self.initialize()
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 437, in initialize
    self._real_initialize()
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/twitch.py", line 58, in _real_initialize
    self._login()
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/twitch.py", line 109, in _login
    redirect_page, handle = login_step(
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/twitch.py", line 90, in login_step
    fail(error)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/twitch.py", line 66, in fail
    raise ExtractorError(
youtube_dl.utils.ExtractorError: Unable to login. Twitch said: Please enter a Login Verification Code.
```

**Comment by cemiu** [comment]
> Here's how youtube-dl does it: https://github.com/ytdl-org/youtube-dl/blob/530f4582d011cd94986cf4d233f9fb9263f72150/youtube_dl/extractor/twitch.py#L60
> 
> Looks a bit janky but doable.

A quick solution would be to just pass along a manually extracted auth_token in the header, to skip the entire login sequence.

Should be as simple as adding
`headers['Authorization'] = f'OAuth {auth_token}'`
to the `authenticated_post` function in `.twitch`, which could be specified through an optional param.

**Comment by ihabunek** [comment]
This is now implemented in [1.22.0](https://github.com/ihabunek/twitch-dl/releases/tag/1.22.0). 

@cemiu thanks for the implementation. I changed the option name to `--auth-token` and added some error handling and more docs.

See docs here:
https://twitch-dl.bezdomni.net/commands/download.html#downloading-subscriber-only-vods

Please test and let me know if it works as expected.
